imsApp/src/imsRecord.cc

```diff
@@ -3589,6 +3647,142 @@ static void post_fields( imsRecord *prec, unsigned short alarm_mask,
     return;
 }
 
+/******************************************************************************/
+static void io_controller ( struct ims_info *mInfo )
+{
+    char *s, *t, buf[2048];
+    const double wtimeout = 1.0;
+    const double rtimeout = 0.1;
+    asynStatus status;
+    int i, eomReason, wild;
+    size_t len;
+
+    for (;;) {
+	/* Check for new work */
+	mInfo->rMutex->lock();
+	for (i = 0; i < REQ_CNT; i++) {
+	    struct ims_asyn_req *req = &mInfo->req[i];
+
+	    if (req->state == IAR_SENDONLY) {
+		pasynOctetSyncIO->write( mInfo->pasynUser, req->out, 
+					 req->outsize, wtimeout, &len );
+		free_request_buf(mInfo, req);
+		continue;
+	    }
+	    if (req->state == IAR_SEND) {
+		pasynOctetSyncIO->write( mInfo->pasynUser, req->out, 
+					 req->outsize, wtimeout, &len );
+		req->state = IAR_RESPWAIT;
+		continue;
+	    }
+	}
+	mInfo->rMutex->unlock();
+
+	/* Try to read a message */
+	status = pasynOctetSyncIO->read( mInfo->pasynUser, buf, sizeof(buf) - 1,
+					 rtimeout, &len, &eomReason );
+	if (status != asynSuccess || len <= 0)
+	    continue;
+
+	if (len < sizeof(buf))
+	    buf[len] = 0;
+
+#if 0
```

Comment:
many really cool logging libraries like spdlog, if we don't want to do debug this way. not for this PR of course